File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-38.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        var res = 0;
        var mostFreq = 0;
        var dict = new Dictionary<char, int>();
        var l = 0;
        var r = 0;
        for (r = 0; r < s.Length; r++) {
            dict[s[r]] = dict.GetValueOrDefault(s[r], 0) + 1;
            mostFreq = Math.Max(mostFreq, dict[s[r]]);
            while ((r - l + 1) - mostFreq > k) {
                dict[s[l]]--;
                l++;
            }
            res = Math.Max(res, r - l + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-38.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)
        res = 0

        while l <= r:
            totalTime = 0
            k = l + ((r - l ) // 2)

            for p in piles:
                totalTime += math.ceil(p / k)
            if totalTime <= h:
                res = k
                r = k - 1
            else:
                l = k + 1
        return res